### PR TITLE
Verify that service cluster is set in case LS tracing is enabled

### PR DESCRIPTION
**Pull request by RomanDzhabarov**

### CODE REVIEW

**Comment by mattklein123** [comment]
test please. you might move the tracing configuration load to a static utility function that is easier to test.

**Comment by moderation** [comment]
@RomanDzhabarov thanks for following up on my testing issue with this PR.

**Comment by RomanDzhabarov** [comment]
Sounds good, will add tests in a bit today.

File: test/server/configuration_impl_test.cc [hunk 1]
```diff
@@ -119,5 +120,38 @@ TEST(ConfigurationImplTest, BadListenerConfig) {
...
+      }
+    }
+  }
+  )EOF";
+
+  Json::ObjectPtr loader = Json::Factory::LoadFromString(json);
+
+  NiceMock<Server::MockInstance> server;
+  std::string cluster_not_set = "";
+  EXPECT_CALL(server.local_info_, clusterName()).WillOnce(ReturnRef(cluster_not_set));
```
Comment (mattklein123) [inline test/server/configuration_impl_test.cc]: Just modify server.local_info_.cluster_name_ directly. You don't need a new variable or the EXPECT_CALL.

File: test/server/configuration_impl_test.cc [hunk 2]
```diff
@@ -119,5 +120,38 @@ TEST(ConfigurationImplTest, BadListenerConfig) {
   EXPECT_THROW(config.initialize(*loader), Json::Exception);
 }
 
+TEST(ConfigurationImplTest, NotSetServiceClusterWhenLSTracing) {
```
Comment (mattklein123) [inline test/server/configuration_impl_test.cc]: ServiceClusterNotSetWhenLSTracing

**Pull request merged**